### PR TITLE
[HttpFoundation] do not access properties before initialization

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 7.0
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | 
| License       | MIT

### CODE REVIEW

**Comment by stof** [comment]
the right solution is actually to add a `null` default value there (otherwise, we break BC for child classes)

**Comment by xabbuh** [comment]
changed it to initialise the properties with `null` instead

File: src/Symfony/Component/HttpFoundation/Request.php [hunk 1]
```diff
@@ -152,13 +152,13 @@ class Request
...
-     * @var array<string, string[]>
+     * @var array<string, string[]>|null
      */
-    protected static array $formats;
+    protected static ?array $formats = null;
 
     protected static ?\Closure $requestFactory = null;
 
-    private string $preferredFormat;
+    private ?string $preferredFormat = null;
```
Comment (stof) [inline src/Symfony/Component/HttpFoundation/Request.php]: I don't think this one needs to be changed. As it is a private method, we can fully control the lazy initialization and the code handling it was fine (as we use `??=`)

Comment (xabbuh) [reply src/Symfony/Component/HttpFoundation/Request.php]: we can keep it as is, but then we need to update `getPreferredFormat()` to not assign `null` to this property (changes pushed)

Comment (stof) [reply src/Symfony/Component/HttpFoundation/Request.php]: ah, I missed that we were sometimes assigning `null`

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**